### ISSUE TITLE
Assertion violation at src/util/old_vector.h:375

### ISSUE THREAD

**Issue by wintered** [Z3Prover/z3#3657]
```
[794] % z3 small.smt2
ASSERTION VIOLATION
File: ../src/util/old_vector.h
Line: 375
idx < size()
(C)ontinue, (A)bort, (S)top, (T)hrow exception, Invoke (G)DB
a
[795] % 
[795] % cat small.smt2
(set-option :unsat_core true)
(declare-fun a () Bool)
(assert (or a a))
(check-sat-using ufbv-rewriter)
[796] %
```

OS: Ubuntu 18.04
Commit: 65b2037

**Comment by rainoftime** [comment]
Hi, seems that you forgot to paste the input

~~~~
(set-option :unsat_core true)
(declare-const v0 Bool)
(declare-const v2 Bool)
(assert  (=> (distinct v0 v2) v0))
(check-sat-using ufbv-rewriter)
~~~~

**Comment by NikolajBjorner** [comment]
fixed with #3705
